src/event/source/unix.rs

```diff
@@ -45,128 +62,98 @@ impl UnixInternalEventSource {
-                                        break;
-                                    }
-                                    // once more data is available to read.
-                                    else if e.kind() == io::ErrorKind::Interrupted {
-                                        continue;
-                                    }
-                                }
-                            };
-
-                            if let Some(event) = self.parser.next() {
-                                return Ok(Some(event));
+            #[cfg(feature = "event-stream")]
+            selector.add(&self.wake_pipe.receiver);
+
+            let _ = selector.select(timeout.leftover())?;
+            if selector.get(&self.tty).is_some() {
+                loop {
+                    match self.tty.read(&mut self.tty_buffer, TTY_BUFFER_SIZE) {
+                        Ok(read_count) => {
+                            if read_count > 0 {
+                                self.parser.advance(
+                                    &self.tty_buffer[..read_count],
+                                    read_count == TTY_BUFFER_SIZE,
+                                );
                             }
                         }
-                    }
-                    SIGNAL_TOKEN => {
-                        for signal in self.signals.pending() {
-                            match signal {
-                                signal_hook::consts::SIGWINCH => {
-                                    // TODO Should we remove tput?
-                                    //
-                                    // This can take a really long time, because terminal::size can
-                                    // launch new process (tput) and then it parses its output. It's
-                                    // not a really long time from the absolute time point of view, but
-                                    // it's a really long time from the mio, async-std/tokio executor, ...
-                                    // point of view.
-                                    let new_size = crate::terminal::size()?;
-                                    return Ok(Some(InternalEvent::Event(Event::Resize(
-                                        new_size.0, new_size.1,
-                                    ))));
-                                }
-                                _ => unreachable!("Synchronize signal registration & handling"),
-                            };
+                        Err(e) => {
+                            match e.kind() {
+                                // No more data to read at the moment. We will receive another event
+                                // once more data is available to read.
+                                io::ErrorKind::WouldBlock => break,
+                                io::ErrorKind::Interrupted => continue,
+                                _ => return Err(e),
```

Comment:
This branch was missing, I believe it was a mistake, but this could result in different behavior